src/lib/cimbar_js/cimbar_js.cpp

```diff
@@ -55,10 +55,24 @@ bool auto_scale_window()
 	return true;
 }
 
+// we may change the api to accept an buff to next_frame()
+// rather than generating a fresh cv::Mat alloc each time
+// but for now, for non-JS purposes we expose this function
+int cimbare_get_frame_buff(unsigned char** buff)
+{
+	if (!_next)
+		return -2;
+	if (_next->cols == 0 or _next->rows == 0)
+		return -1;
+
+	*buff = _next->data;
+	return _next->cols * _next->rows * _next->channels();
+}
+
 // render() and next_frame() could be put in the same function,
 // but it seems cleaner to split them.
 // in any case, we're concerned with frame pacing (some encodes take longer than others)
```

Comment:
I *think* this comment is less relevant as the scope of this api slowly grows. Having them separate is clearly the right way imo